2_gitlab/create-repos.sh

```diff
@@ -29,6 +29,7 @@ pushd gitlab-repos
     for cluster in ${CLUSTERS}; do
        test -f ${cluster} || ssh-keygen -f ${cluster} -N ''
        # Create a secret with the private key
+       gcloud secrets delete gitlab-cluster-key-${cluster} --quiet || true
```

Comment:
We should probably only do this in destroy, we don't want to delete these if someone already has them (but didn't destroy).

Reply:
new keys are created every time (line 30) so we need to delete the existing key and replace with a new.  Perhaps we can do versions instead, but the value does need to be replaced

Reply:
Every time you rebuild the infra? Wouldn't destroy take care of that though?

If you think this is the right approach, that's fine though.

Reply:
they are recreated in the very next line.. the script regenerates the keys and there is no api to update a secret version, only delete and recreate. 
thanks

Reply:
I'd like to keep this as-is for now.  Destroy does contain a removal of the secret. This is mostly used as a protection to ensure the key is stored in the event that the secret exists (providing idempotency)